### PR TITLE
Use MessageBuffer in RecoveryPermitter, #28417

### PR DESCRIPTION
**Pull request by patriknw**
* to propagate trace context

Refs https://github.com/akka/akka/issues/28417

### CODE REVIEW

**Review by raboof — APPROVED** [review]

**Comment by patriknw** [comment]
let's wait with merging until someone from Cinnamon team has looked

**Review by ygree — APPROVED** [review]
LGTM. Thanks! Not super urgent but would be great to backport it to 2.5 as well.

File: akka-actor/src/main/scala/akka/util/MessageBuffer.scala [hunk 1]
```diff
@@ -109,13 +110,49 @@ final class MessageBuffer private (private var _head: MessageBuffer.Node, privat
...
+      _tail = null
+    } else if (_size == 1) {
+      _head = result.head
+      _tail = result.head
+      _tail.next = null
+    } else {
+      _head = result.head
+      _tail = result.last
+      _tail.next = null
+      result.zipWithIndex.foreach {
```
Comment (hepin1989) [inline akka-actor/src/main/scala/akka/util/MessageBuffer.scala]: remove the zipWithIndex with plain while?

Comment (patriknw) [reply akka-actor/src/main/scala/akka/util/MessageBuffer.scala]: changed to while

**Pull request merged**